### DISCUSSION TITLE
[datasource ] loki log explore : Please add a switch to turn off "full-range log volume"

### DISCUSSION THREAD

**Discussion by liguozhong** [grafana/grafana#63695] [Data Sources]
![image](https://user-images.githubusercontent.com/9583245/221113398-000ad2b0-f78b-4a38-bbec-c0936210bc45.png)

After we upgraded to grafana 9.0, our log explore became very difficult to use, because we have a large（700Mb/s） number of logs, and this trend graph（metrics query） cannot be returned . 

We are upgrading from grafana 7.5 to 9.0, please provide a switch to turn off the "full-range log volume" feature.

We're really in big trouble, now we can't go back to 7.5

**Comment by liguozhong** [comment]
@gabor  @ivanahuckova 
Please help us solve this loki problem, we switch from ELK to loki, there are a lot of queries every day for the traceID keyword, we don’t need to see this log volumn trend.

**Reply by gabor** [reply]
hi @liguozhong , you can collapse the logs-volume "box" if you do not want to see it, and then it will not run the associated Loki query. does that solve your problem?

**Comment by liguozhong** [comment]
thanks ！ I didn't see this button in version 9.0.  which version can see this switch（collapse the logs-volume "box"） to cancel the full log volume.
![image](https://user-images.githubusercontent.com/9583245/221564372-ef569876-2ab5-4a47-b9a6-b13685803972.png)

**Comment by liguozhong** [comment]
If this （collapse the logs-volume "box"） can cancel the log volume query, it can solve my problem .

**Comment by liguozhong** [comment]
thanks !!!!!，I see it in main branch.
![image](https://user-images.githubusercontent.com/9583245/221738103-0e96f158-4ed8-4520-9380-3f9c32929ee9.png)

**Comment by liguozhong** [comment]
thanks @gabor ,I have another suggestion:


whether it is possible to optimize the existing logic, wait for the `table log query` to return the result, if the return result of the` table log query `is less than `limit: 1000`, then the histogram of the `log volume` use the result of the log query as input instead of Initiate a `metrics query` again. 

why?
Because a large number of our queries are less than the limit, but these queries are all traceID queries(`{app="buy2"} | "traceID or orderId"`), and the performance is very slow. is an optimization. 

please consider it .

I gave your suggestion back to my sre team, but my team is still not satisfied. They still want to see the histogram distribution of traceID log query. Maybe this suggestion is a compromise and an optimization of loki.

**Comment by mary-sitzenstatter** [comment]
@radiorental @niat22

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). The original content of the discussion seems to have been addressed, but there seems to be another feature request in the comments. Please feel free to make an [issue](https://github.com/grafana/grafana/issues) if you cannot find a solution and would like to request that feature.
